File: 12월 3주차/aoreo/02/[프로그래머스] 카펫.java
class Solution {
  public int[] solution(int brown, int yellow) {
      int[] answer = new int[2];
      int cnt = 1;
      while (yellow / cnt >= cnt) { 
        if(yellow % cnt == 0) {
            if(((yellow/cnt)+2)*(cnt+2) - yellow == brown){
              answer[0] = (yellow/cnt)+2;
              answer[1] = cnt + 2;
              break;
            }
        }
        cnt += 1;
      }
      return answer;
  }
}

File: HongJun/01/[프로그래머스]카펫.kt
class Solution {
    fun solution(brown: Int, yellow: Int): IntArray {
        val answer = IntArray(2)

        // 전체 넓이
        val total = brown + yellow

        // 세로 길이는 최소 3부터 시작
        for (h in 3..total) {

            // 세로 길이가 전체 넓이의 약수인 경우만 고려
            if (total % h == 0) {

                // 가로 길이 구하기
                var w = total/h

                // 노랑 개수 맞추기
                if ((w-2)*(h-2) == yellow) {

                    // 가로가 세로보다 같거나 긴 경우만 고려
                    if (w >= h) {
                        answer[0] = w
                        answer[1] = h
                    }

                }
            }
        }

        return answer
    }
}